Makefile

```diff
@@ -103,7 +103,7 @@ stack-install-test :
 
 # Copy the artefacts built by Stack as if they were build by Cabal.
 .PHONY : stack-copy-artefacts
-stack-copy-artefacts : stack-install-bin stack-install-test
+stack-copy-artefacts : 
```

Comment:
Not sure what the author of this goal `stack-copy-artefacts` had in mind, so I am not sure it is ok to remove the prerequisites.  (I am not a stack user.)
It does not seem to fit into this PR, which is about a semantics-preserving polishing of Haskell code.

Reply:
Ah, indeed. 

I was annoyed by this part. The testing suite depends on `Agda` (in `Agda.cabal`), so `stack-install-test` usually re-compiles Agda from scratch even `stack-install-bin` has been done beforehand. So, `make install-bin` with `stack.yaml` usually compiles Agda 4 times and retrieve the last only. I will create another PR to fix this.